tests/test_query_engine.py

```diff
@@ -1023,3 +1023,46 @@ class Cohort:
         dict(patient_id=3, has_code="yes"),
         dict(patient_id=4, has_code="yes"),
     ]
+
+
+@pytest.mark.integration
+def test_categorise_multiple_truthiness_categories(database, setup_test_database):
```

Comment:
This tests for the sorting bug above

Reply:
We could really do with unit tests for this kind of case, couldn't we? It's very hard to see what's really being tested for. Not asking you to add one now, but we need to work out how to do it at some point.

Reply:
For now I'll add a comment about what it's testing